lib/lpc55-rot-startup/src/lib.rs

```diff
@@ -74,6 +75,13 @@ fn puf_check(puf: &lpc55_pac::PUF) {
     }
 }
 
+/// Run the common startup routine for LPC55-based roots of trust.
+///
+/// NOTE: this function is marked `inline(never)` to try to ensure that its
+/// stack usage is not commingled with that of the calling routine, to enable
+/// the use of `nuke_stack` to make extra sure that no data is left in memory by
+/// this routine.
+#[inline(never)]
```

Comment:
Not thrilled with relying on this attribute to enforce that separation.  Would moving `dice::run` to an software-triggered interrupt let us play any games with ensuring it uses a different stack that we can sanitize afterward?